torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -624,13 +701,28 @@ ModuleMethodVector InsertQuantDeQuantHelper::getInvokedMethods(
   return invoked_methods;
 }
 
-void propagateQParams(
+void InsertQuantDeQuantHelper::propagateQParams(
     Value* original_output,
     const std::vector<Value*>& inputs,
-    const c10::optional<std::tuple<c10::QScheme, QParamVector>>& qparams_opt =
-        c10::nullopt) {
+    bool is_scalar,
+    const c10::optional<std::tuple<c10::QScheme, QParamVector>>& qparams_opt) {
   Node* n = original_output->node();
   Graph* graph = n->owningGraph();
+  if (is_scalar) {
+    // convert Scalar to Tensor
```

Comment:
Might be useful to make a scalar to tensor and tensor to scalar conversion function for this snippet and reuse it across ops.